x-pack/plugins/maps/public/ems_autosuggest/ems_autosuggest.ts

```diff
@@ -20,65 +18,36 @@ export interface EMSTermJoinConfig {
   field: string;
 }
 
-const wellKnownColumnNames = [
-  {
-    regex: /(geo\.){0,}country_iso_code$/i, // ECS postfix for country
-    emsConfig: {
-      layerId: emsWorldLayerId,
-      field: 'iso2',
-    },
-  },
-  {
-    regex: /(geo\.){0,}region_iso_code$/i, // ECS postfixn for region
-    emsConfig: {
-      layerId: emsRegionLayerId,
-      field: 'region_iso_code',
-    },
-  },
-  {
-    regex: /^country/i, // anything starting with country
-    emsConfig: {
-      layerId: emsWorldLayerId,
-      field: 'name',
-    },
-  },
-];
-
-const wellKnownColumnFormats = [
-  {
-    regex: /(^\d{5}$)/i, // 5-digit zipcode
-    emsConfig: {
-      layerId: emsUsaZipLayerId,
-      field: 'zip',
-    },
-  },
-];
-
 interface UniqueMatch {
   config: { layerId: string; field: string };
   count: number;
 }
+interface FileLayerFieldShim {
+  id: string;
+  values?: string[];
+  regex?: string;
+  alias?: string[];
+}
+
+type SampleValues = Array<string | number>;
```

Comment:
I would not recommend typing SampleValues. Its hard to remember what the type is from the name and does not hide a lot of complexity. How about just using "Array<string | number>" instead of SampleValues?